src/main/java/marquez/client/models/Dataset.java

```diff
@@ -42,15 +43,21 @@
   @Getter @NonNull private final Instant createdAt;
   @Getter @NonNull private final Instant updatedAt;
   @Getter @NonNull private final String sourceName;
-  @Nullable private final Instant lastModified;
+  @Nullable private final List<Field> fields;
+  @Nullable private final List<String> tags;
+  @Nullable private final Instant lastModifiedAt;
   @Nullable private final String description;
 
+  public Optional<List<Field>> getFields() {return Optional.ofNullable(fields);}
+
+  public Optional<List<String>> getTags() {return Optional.ofNullable(tags);}
```

Comment:
Returning an _empty_ `List` is preferred over wrapping the returned `List` in an `Optional` and, I feel, provides clearer semantics:

> Container types, including collections, maps, streams, arrays, and optionals should not be wrapped in optionals. Rather than returning an empty Optional<List<T>>, you should simply return an empty List<T>

**see**, [Item 54: Return empty collections or arrays, not nulls](https://github.com/nicolasmanic/effective-java-in-a-nutshell#item-55-return-optionals-judiciously)

Let's handle the case where the _fields_ or _tags_ can be null by defining  the constructor (I left out all other fields for readability):

```java
public Dataset(
  @Nullable final List<Field> fields,
  @Nullable final List<String> tags) {
  this.fields = ImmutableList.copyOf(new ArrayList<>(fields)); 
  this.tags = ImmutableList.copyOf(new ArrayList<>(tags)); 
}
```

Reply:
Done